lib/ancestry/class_methods.rb

```diff
@@ -206,5 +206,14 @@ def unscoped_where
         yield self.ancestry_base_class.unscope(:where)
       end
     end
+
+    ANCESTRY_UNCAST_TYPES = [:string, :uuid, :text].freeze
+    def primary_key_is_an_integer?
```

Comment:
I really like what you ended up with here.  👍

Reply:
Though, looking at the test failures, you might have to do something like this if we are still supporting Rails 3:

```ruby
if Rails.version < "4.0"
  def primary_key_is_an_integer?
    if defined?(@primary_key_is_an_integer)
      @primary_key_is_an_integer
    else
      @primary_key_is_an_integer = !ANCESTRY_UNCAST_TYPES.include?(columns_hash[primary_key.to_s].type)
    end
  end
else
  # The method definition you have here
end
```

The `columns_hash[primary_key.to_s].type` is the class method equivalent of `column_for_attribute`.

Actually, you should just do that for all versions, since it seems to exist in Rails 5 as well.

Reply:
I started to change it but just using the pre 4.0 version:

```ruby
    ANCESTRY_UNCAST_TYPES = [:string, :uuid, :text].freeze
    if ActiveSupport::VERSION::STRING < "4.0"
      def primary_key_is_an_integer?
        if defined?(@primary_key_is_an_integer)
          @primary_key_is_an_integer
        else
          @primary_key_is_an_integer = !ANCESTRY_UNCAST_TYPES.include?(column_for_attribute(primary_key).type)
        end
      end
    else
      def primary_key_is_an_integer?
        if defined?(@primary_key_is_an_integer)
          @primary_key_is_an_integer
        else
          @primary_key_is_an_integer = !ANCESTRY_UNCAST_TYPES.include?(type_for_attribute(primary_key))
        end
      end
    end
```